php_phongo_structs.h

```diff
@@ -127,6 +127,12 @@ typedef struct {
 	zend_object              std;
 } php_phongo_session_t;
 
+typedef struct {
+	mongoc_topology_description_t* topology_description;
+	HashTable*                     properties;
+	zend_object                    std;
+} php_phongo_topologydescription_t;
+
```

Comment:
Should I leave this as it is now (modeled after `php_phongo_server_description_t`) or should I still change it to hold the topology `type` and `server_descriptions` as we discussed yesterday?

Reply:
@kevinAlbs confirmed in Slack that `mongoc_topology_description_t` _doesn't_ actually depend on a libmongoc client (ignoring the APM callbacks/context property, which we can guarantee won't be used for PHP's TopologyDescription object). This can remain as-is.

That means your init method for the TopologyDescription class will be more straightforward, and complexity essentially shifts to the getType() and getServerDescriptions() methods.